uobtheatre/bookings/schema.py

```diff
@@ -455,21 +456,77 @@ def resolve_mutation(cls, _, info, booking_reference, tickets, performance_id):
             # Check the ticket booking matches the given booking
             if ticket.booking != booking:
                 raise GQLFieldException(
+                    field="booking_reference",
                     message="The ticket booking does not match the mutation booking."
                 )
 
+            if ticket.checked_in:
+                raise GQLNonFieldException(
+                    message="Ticket ID {} is already checked in".format(ticket.id)
+                )
+
             ticket.check_in()
-            # Raise ticket booking does not match the booking
 
         return CheckInBooking(booking=booking, performance=performance)
 
+class UnCheckInBooking(AuthRequiredMixin, SafeMutation):
+    """Mutation to un-check in the tickets of a Booking.
+
+    Args:
+        booking_reference (str): The reference for the Booking being paid for.
+        performance_id (str): The id of the performance that the ticket is
+            being booked in for, this should match the performance of the
+            booking. If this is not the case an error will be thrown as the
+            Booking cannot be used for the Performance.
+
+
+    Returns:
+        booking (BookingNode): The Booking which was paid for.
+        payment (PaymentNode): The Payment which was created by the
+            transaction.
+
+    Raises:
+        GQLNonFieldException: If the un-check in was unsuccessful
+    """
+
+    performance = graphene.Field("uobtheatre.productions.schema.PerformanceNode")
+    booking = graphene.Field(BookingNode)
+
+    class Arguments:
+        booking_reference = graphene.String(required=True)
+        performance_id = IdInputField(required=True)
+        tickets = graphene.List(CheckInTicketInput, required=True)
+
+    @classmethod
+    def resolve_mutation(cls, _, info, booking_reference, tickets, performance_id):
+        performance = Performance.objects.get(id=performance_id)
+        booking = Booking.objects.get(reference=booking_reference)
+
+        # check if the booking pertains to the correct performance
+        if booking.performance != performance:
+            raise GQLFieldException(
+                message="The booking performance does not match the given performance."
+            )
+            # raise booking performance does not match performance given
+        ticket_objects = []
+
+        ticket_objects = list(map(lambda ticket: ticket.to_ticket(), tickets))
+        # loop through the ticket IDs given
+        for ticket in ticket_objects:
+            # Check the ticket booking matches the given booking
+            if ticket.booking != booking:
+                raise GQLFieldException(
+                    message="The ticket booking does not match the mutation booking."
+                )
+
+            ticket.uncheck_in()
```

Comment:
If there is an error where a ticket is not in the booking then I assume we want to not check in any of the tickets rather than just stopping halfway through.  You might be able to fix this by wrapping in atomic transaction.

Reply:
Its probably worth also writing a test for this.